### PR TITLE
unix: Enable extra compiler warnings.

### PR DESCRIPTION
**Pull request by dpgeorge**
Although we are not enabling all warnings, this will be enough to close issue #699.

The reason it's a PR is that it requires changes to 3rd-party code in extmod.  @pfalcon can we make the changes upstream?

### CODE REVIEW

**Comment by pfalcon** [comment]
@dpgeorge: Ack. I'll look into merging as many as possible of these fixes to upstream project(s) (for what I'd like to understand them first). Hope this is not top priority and can wait, because I have more important (like "uzlib doesn't work at all" https://github.com/micropython/micropython/commit/e6a118ee85b35eb443057a2b6dbc87059deff6dc) and more interesting changes in the queue ;-).

**Comment by dpgeorge** [comment]
@pfalcon ok.  Not a priority, just want to eventually close #699.

**Comment by dpgeorge** [comment]
@pfalcon ping :)

**Comment by pfalcon** [comment]
Ah, thanks for reminder, will work on this coming days.

File: extmod/re1.5/compilecode.c [hunk 1]
```diff
@@ -66,7 +66,7 @@ int re1_5_sizecode(const char *re)
 
 #define EMIT(at, byte) code[at] = byte
 
-const char *_compilecode(const char *re, ByteProg *prog)
+static const char *_compilecode(const char *re, ByteProg *prog)
```
Comment (pfalcon) [inline extmod/re1.5/compilecode.c]: Merged to upstream and propagated to upy.

File: extmod/uzlib/tinf.h [hunk 2]
```diff
@@ -76,7 +76,7 @@ int TINFCC tinf_zlib_uncompress_dyn(TINF_DATA *d, unsigned int sourceLen);
 
 /* high-level API */
 
-void TINFCC tinf_init();
+void TINFCC tinf_init(void);
```
Comment (pfalcon) [inline extmod/uzlib/tinf.h]: uzlib changes pushed to upstream and propagated to upy.

**Comment by pfalcon** [comment]
So, I merged changes to upstreams I maintain, so feel free to rebase and commit rest of patch.

**Comment by dpgeorge** [comment]
Thanks.  Merged in 92496abe0f4e606a5fd8227d861aedff466b9679.

**Pull request closed without merging**